File: src/main/java/com/coffeeshop/model/entity/productDetails/Product.java
package com.coffeeshop.model.entity.productDetails;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder

@Entity
@Table(name="Products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;

    @NotBlank
    private String productName;

    @NotNull
    private Integer quantityAvailableKg;

    private String[] productImages;

    @NotBlank
    private String description;

    @NotNull
    private Double unitPrice;
}

File: ui/customer-ui/src/app/checkout-customer/model/product.ts
 export class Product {

   private _productId: number;
   private _quantity: number;

   get productId(): number {
     return this._productId;
   }

   set productId(value: number) {
     this._productId = value;
   }

   get quantity(): number {
     return this._quantity;
   }

   set quantity(value: number) {
     this._quantity = value;
   }

   constructor(productId: number, quantity: number) {
     this._productId = productId;
     this._quantity = quantity;
   }


 }
